rfcs/inline_symbols.md

```diff
@@ -0,0 +1,624 @@
+be positioned over the annotated value, which it can read or skip as needed.
+
+##### `0xE2`: Multiple inline symbol annotations
+
+```
+            7       4 3       0
+            +---------+---------+
+Value with  |   14    |    2    |
+multiple    +---------+---------+=======================================+
+annotations :     length [VarUInt]                                      :
+            +-----------------------------------------------------------+
+            :     annotation [VarInt + optional UTF8 representation]    : ...
+            +-----------------------------------------------------------+
+            :     value                                                 :
+            +-----------------------------------------------------------+
+```
+
+Unlike the singleton annotation encoding, the encoding for a value with multiple annotations
+includes a `length` field that indicates the number of bytes that will be used to represent
+the sequence of annotations that follow. It does not include the length of the value itself,
+which provides a length encoding of its own.
+
+Annotations in the sequence do not need to be encoded homogenously; writers can write some
+annotations as symbol IDs and others as inline text. (See the section [*Inline symbol
+structs*](#0xf4-inline-symbol-structs) for more detail.)
+
+To skip an annotation sequence, the reader must read the `VarUInt` `length` and skip that number of
+bytes. The reader will then be positioned over the annotated value, which it can read or skip as
+needed.
+
+### Inline symbols' relationship to Shared Symbol Tables
+
+Ion 1.0 provides [Shared Symbol
+Tables](http://amzn.github.io/ion-docs/docs/symbols.html#shared-symbol-tables) (SSTs)
+as a mechanism for reducing the overhead of defining a set of symbols in a stream.
+
+Here we revisit our example weather station data from the [Short-lived
+streams](#short-lived-streams) section, which used a Local Symbol Table to define all of the symbols
+it needed at the beginning of the stream:
+
+```js
+$ion_1_0
+$ion_symbol_table::{ // Add all of our symbols, field names, and annotations to the symbol table
+  sensorId,   // $10
+  type,       // $11
+  sensorData, // $12
+  reading,    // $13
+  temperature,// $14
+  celsius,    // $15
+  time,       // $16
+}
+{
+  $10: 12345, // Then encode our symbols, field names, and annotations as symbol IDs
+  $11: $12,
+  $13: {
+    $14: $15::12.5,
+	$16: 2020-10-22T16:00:00Z
+  }
+}
+```
+
+Because this struct layout is going to be reused frequently, the weather service could have instead
+created an SST with all of the necessary symbols:
+
+```js
+$ion_1_0
+$ion_shared_symbol_table::{
+  name: "com.example.weather.symbols",
+  version: 1,
+  symbols: [ // Create a Shared Symbol Table with the necessary symbols
+    sensorId,
+    type,
+    sensorData,
+    reading,
+    temperature,
+    celsius,
+    time,
```

Comment:
Should be string values, unless we allow inline symbols here...

Reply:
Fixed, thanks.